### DISCUSSION TITLE
memory increase when create swapchain

### DISCUSSION THREAD

**Discussion by simonsland** [google/filament#5525] [Q&A]
When I create a swapChain by pass a CAMetalLayer at iphoneX，the memory usage increase 100+MB，which confuse me why it takes so much memory. And the size of CAMetalLayer created is the bounds of screen.

**Comment by bejado** [comment]
My hunch is that the memory increase is due to Filament internal textures being allocated. Some things you can test to see how the memory usage is affected:
1. Pass a smaller `CAMetalLayer` (not full screen).
2. Try using a headless `SwapChain`.
3. Turn off all post-processing effects.

**Reply by caorm** [reply]
@bejado 

> 2\. Try using a headless `SwapChain`.

engine->createSwapChain(1242, 2208, SwapChain::CONFIG_TRANSPARENT);

How to draw the model to the View?  I don't understand how to use a headless `SwapChain`.

Can you clarify?  thank you！

**Reply by bejado** [reply]
A headless SwapChain can be used like a regular `SwapChain`, except behind the scenes a texture (instead of a `CAMetalDrawable`) is allocated and the results aren't shown on the screen. I meant this as a test to see how the memory usage is affected.
